i18n(fr): Adding `imports.md` translations (#502)
withastro/docs: src/pages/fr/guides/imports.md

Alors que Astro comprend la syntaxe JSX par défaut, vous aurez besoin d'inclure une intégration de Framework pour les afficher, comme React, Preact et Solid. Consultez notre [Guide d'utilisation des intégrations](/fr/guides/integrations-guide/) pour en savoir plus.

**Note: Astro ne supporte pas la syntaxe JSX dans les fichiers `.js`/`.ts`.** JSX ne sera géré que dans les fichiers qui se terminent par l'extension `.jsx` et `.tsx`.

## Packages NPM

```js
// Retourne les packages NPM React & React-DOM
import React from 'react';
import ReactDOM from 'react-dom';
```

Astro vous permet d'importer des packages NPM directement dans le navigateur. Même si un package a été publié en format legacy, Astro va convertir le package en ESM avant de le fournir au navigateur.

## JSON

```js
// Charge l'objet JSON via l'export par défaut
import json from './data.json';
```

Astro supporte l'importation directe des fichiers JSON dans votre application. Les fichiers importés retournent l'objet JSON complet dans l'import par défaut.

## CSS

```js
// Charge et intègre 'style.css' sur la page
import './style.css';
```

> Astro supporte l'importation directe des fichiers CSS dans votre application. Les styles importés n'exposent aucun export, mais l'importation d'un fichier CSS le rendra automatiquement disponible sur la page. Cela fonctionne par défaut pour tous les fichiers CSS, et peut supporter les langages de compilation CSS comme Sass et Less via des plugins.
> 
> Si vous préférez ne pas écrire de CSS, Astro supporte également tous les librairies "CSS-in-JS" populaires (ex: styled-components) pour styliser vos éléments.

Reviewer: ```suggestion
Astro supporte l'importation de fichiers CSS directement dans votre application. Les styles importés n'exposent aucune exportation, mais l'importation d'un style ajoutera automatiquement ces styles à la page. Cela fonctionne pour tous les fichiers CSS par défaut, et peut prendre en charge les langages de compilation vers le CSS comme Sass et Less via des plugins.

Si vous préférez ne pas écrire de CSS, Astro prend également en charge toutes les bibliothèques CSS-in-JS populaires (ex : styled-components) pour le style.
```

## Modules CSS

```jsx
// 1. Convertit les noms de classes './style.module.css' en valeurs uniques, portées.
// 2. Retourne un objet qui associe les noms de classes d'origine à leur valeur portée finale.
import styles from './style.module.css';

// Cet exemple utilise JSX, mais vous pouvez utiliser les modules CSS avec n'importe quel Framework.
return <div className={styles.error}>Votre message d'erreur</div>;
```

> Astro supporte les modules CSS en utilisant la convention de nommage `[name].module.css`. Comme tous les fichiers CSS, l'importation d'un fichier CSS le rendra disponible sur la page. Cependant, les modules CSS exportent un objet spécial nommé `styles` qui associe les noms de classes d'origine à des identifiants uniques.
> 
> Les modules CSS vous aident à garantir l'isolation et la portée des composants sur le frontend avec des noms de classes uniques pour vos feuilles de styles.

Reviewer: ```suggestion
Astro supporte les modules CSS en utilisant la convention de dénomination `[nom].module.css`. Comme tout fichier CSS, l'importation d'un module CSS entraîne l'application automatique de ce CSS à la page. Toutefois, les modules CSS exportent un objet `styles` spécial par défaut qui transforme vos noms de classe originaux en identifiants uniques.

Les modules CSS vous aident à renforcer la délimitation et l'isolation des composants sur le front-end grâce à des noms de classe générés de manière unique pour vos feuilles de style.
```

## Autres ressources

```jsx
import imgReference from './image.png'; // img === '/src/image.png'
import svgReference from './image.svg'; // svg === '/src/image.svg'
import txtReference from './words.txt'; // txt === '/src/words.txt'

// Cet exemple utilise JSX, mais vous pouvez utiliser des références d'importation avec n'importe quel Framework.
<img src={imgReference} />;
```

> Toutes les autres ressources non mentionnées ci-dessus peuvent être importées via l'`import` ESM et retournera une référence URL vers l'asset final construit. Cela peut être utile pour référencer des ressources non-JS par URL, comme créer un élément image avec un attribut `src` pointant vers cette image.
> 
> Il peut aussi être utile de placer des images dans le dossier `public/` comme expliqué sur la [page de structure du projet](/fr/core-concepts/project-structure/#public).

Reviewer: ```suggestion
Toutes les autres ressources qui ne sont pas explicitement mentionnés ci-dessus peuvent être importés via ESM `import` et retourneront une référence URL vers la ressource finale construit. Cela peut être utile pour référencer des ressources non-JS par URL, comme créer un élément image avec un attribut `src` pointant vers cette image.

Il peut également être utile de placer les images dans le dossier `public/` comme expliqué sur la page [page de structure du projet](/fr/core-concepts/project-structure/#public).
```

## WASM

```js
// Charge et initialise le fichier WASM demandé
const wasm = await WebAssembly.instantiateStreaming(fetch('/example.wasm'));
```

> Astro supporte l'importation directe des fichiers WASM dans votre application en utilisant l'API [`WebAssembly`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) du navigateur.

Reviewer: ```suggestion
Astro supporte le chargement de fichiers WASM directement dans votre application en utilisant l'API [`WebAssembly`] (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) du navigateur.
```

## Intégrations Node

> Nous encourageons les utilisateurs d'Astro à éviter les intégrations Node.js (`fs`, `path`, etc) lorsque possible. Astro à comme objectif de rester compatible avec plusieurs environnements de développement JavaScript dans le futur. Cela inclut [Deno](https://deno.land/) et [Cloudflare Workers](https://workers.cloudflare.com/) qui ne supportent pas les modules intégrés Node.js tels que `fs`.
> 
> Notre objectif est de fournir des alternatives aux intégrations Node.js les plus courantes. Cependant, aucune alternative n'existe actuellement. Si vous avez _vraiment_ besoin de ces intégrations, nous n'allons pas vous arrêter. Astro supporte les intégrations Node.js en utilisant le préfixe `node:` de Node.js. Si vous souhaitez lire un fichier, par exemple, vous pouvez le faire de cette manière :

Reviewer: ```suggestion
Nous encourageons les utilisateurs d'Astro à éviter les builtins Node.js (`fs`, `path`, etc.) autant que possible. Astro a pour objectif d'être compatible avec plusieurs environnements de développement. Cela inclut [Deno](https://deno.land/) et [Cloudflare Workers](https://workers.cloudflare.com/) qui ne prennent pas en charge les modules intégrés de Node tels que `fs`.

Notre objectif est de fournir des alternatives aux modules intégrés courants de Node.js. Cependant, aucune alternative de ce type n'existe aujourd'hui. Donc, si vous avez _vraiment_ besoin d'utiliser ces modules builtins, nous ne voulons pas vous en empêcher. Astro supporte les modules builtins de Node.js en utilisant le nouveau préfixe `node:` de Node. Si vous voulez lire un fichier, par exemple, vous pouvez le faire comme ceci :
```

Author: "builtins" serais alors a noter comme mot a ne pas traduire ? a ce moment là on le traite comme un nom propre

Le préfixe `node:` n'est pas nouveau et existe depuis la création de node, il est juste optionnel, mais demandé pour éviter les erreur d'importations

```suggestion
Nous encourageons les utilisateurs d'Astro à éviter les "Built-in" Node.js (`fs`, `path`, etc.) autant que possible. Astro a pour objectif d'être compatible avec plusieurs environnements de développement. Cela inclut [Deno](https://deno.land/) et [Cloudflare Workers](https://workers.cloudflare.com/) qui ne prennent pas en charge les modules intégrés de Node tels que `fs`.

Notre objectif est de fournir des alternatives aux modules intégrés courants de Node.js. Cependant, aucune alternative de ce type n'existe aujourd'hui. Donc, si vous avez _vraiment_ besoin d'utiliser ces modules Built-in, nous ne voulons pas vous en empêcher. Astro supporte les modules Built-in de Node.js en utilisant le préfixe `node:` de Node. Si vous voulez lire un fichier, par exemple, vous pouvez le faire comme ceci :
```
